packages/react-components/src/components/utils/useLongPress.ts

```diff
@@ -29,38 +29,48 @@ export default function useLongPress(
     setIsLongPress(false);
     timerRef.current = setTimeout(() => {
       setIsLongPress(true);
+      onLongPress();
     }, 500);
   }
 
   function handleOnClick(): void {
-    // when it's mobile use ontouchstart and ontouchend to fire onclick and onlongpress event
     if (isMobile) {
       return;
     }
-    onClick();
-    if (isLongPress) {
-      onLongPress();
-      return;
+    if (!isLongPress) {
+      onClick();
     }
   }
 
   function handleOnKeyDown(): void {
+    if (isMobile) {
+      return;
+    }
     if (action) {
       setAction(false);
       startPressTimer();
```

Comment:
(existing bug in this hook)
I think this means we could start two timers, if I mouseDown on something and before mouseUp I activate the keyDown it will trigger a new timer with somewhat undefined behavior on the previous timer

Reply:
This scenarios will end up calling onLongPress twice. I think that's the expected behaviour given that the user explicitly provided 2 inputs.